src/rgw/rgw_rest_s3.cc

```diff
@@ -5870,3 +5875,322 @@ bool rgw::auth::s3::S3AnonymousEngine::is_applicable(
 
   return route == AwsRoute::QUERY_STRING && version == AwsVersion::UNKNOWN;
 }
+
+
+using namespace s3selectEngine;
+const char* RGWSelectObj_ObjStore_S3::header_name_str[3] = {":event-type", ":content-type", ":message-type"};
+const char* RGWSelectObj_ObjStore_S3::header_value_str[3] = {"Records", "application/octet-stream", "event"};
+
+RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3()
+{
+  s3select_syntax = std::make_unique<s3selectEngine::s3select>(),
+  m_s3_csv_object = std::unique_ptr<s3selectEngine::csv_object>(),
+  m_buff_header = std::make_unique<char[]>(1000),
+  chunk_number = 0,
+  crc32 = std::unique_ptr<boost::crc_32_type>(),
+
+  set_get_data(true);
```

Comment:
please replace these assignments with the constructor member initializers:
```suggestion
RGWSelectObj_ObjStore_S3::RGWSelectObj_ObjStore_S3()
  : s3select_syntax(std::make_unique<s3selectEngine::s3select>()),
    m_buff_header(std::make_unique<char[]>(1000)),
    chunk_number(0)
{
  set_get_data(true);
```